pkg/api/v1/clients/memory/resource_client_test.go

```diff
@@ -62,6 +62,178 @@ var _ = Describe("Base", func() {
+			}
+		})
+
+		It("lists resources in a given namespace", func() {
+			resources, err := client.List("ns2", clients.ListOpts{})
+			Expect(err).NotTo(HaveOccurred())
+
+			expectedResourceNames := []string{
+				"name2", "name4",
+			}
+			Expect(resources).To(HaveLen(len(expectedResourceNames)))
+			for i, r := range resources {
+				Expect(r.GetMetadata().GetName()).To(Equal(expectedResourceNames[i]))
+			}
+		})
+
+		It("returns empty list if namespace is invalid", func() {
+			resources, err := client.List("invalid-namespace", clients.ListOpts{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resources).To(HaveLen(0))
+		})
+
+		It("returns resources matching the given selector, across all namespaces", func() {
+			resources, err := client.List("", clients.ListOpts{
+				Selector: map[string]string{
+					"key": "val2",
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// resources are sorted by namespace, then name
+			expectedResourceNames := []string{
+				"name3", "name2",
+			}
+			Expect(resources).To(HaveLen(len(expectedResourceNames)))
+			for i, r := range resources {
+				Expect(r.GetMetadata().GetName()).To(Equal(expectedResourceNames[i]))
+			}
+		})
+
+		It("returns resources matching the given selector, in given namespace", func() {
+			resources, err := client.List("ns2", clients.ListOpts{
+				Selector: map[string]string{
+					"key": "val2",
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(resources).To(HaveLen(1))
+			Expect(resources[0].GetMetadata().GetName()).To(Equal("name2"))
+		})
+
+		It("returns resources matching the given expression selector, across all namespaces", func() {
+			resources, err := client.List("", clients.ListOpts{
+				ExpressionSelector: "key in (val1,val3)",
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// resources are sorted by namespace, then name
+			expectedResourceNames := []string{
+				"name1", "name5", "name4",
+			}
+			Expect(resources).To(HaveLen(len(expectedResourceNames)))
+			for i, r := range resources {
+				Expect(r.GetMetadata().GetName()).To(Equal(expectedResourceNames[i]))
+			}
+		})
+
+		It("returns resources matching the given expression selector, in given namespace", func() {
+			resources, err := client.List("ns2", clients.ListOpts{
+				ExpressionSelector: "key in (val1,val3)",
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(resources).To(HaveLen(1))
+			Expect(resources[0].GetMetadata().GetName()).To(Equal("name4"))
+		})
+
+		It("when both selector and expression selector are provided, uses expression selector", func() {
+			resources, err := client.List("ns2", clients.ListOpts{
+				Selector: map[string]string{
+					"ignored": "selector",
+				},
+				ExpressionSelector: "key in (val1,val3)",
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(resources).To(HaveLen(1))
+			Expect(resources[0].GetMetadata().GetName()).To(Equal("name4"))
+		})
+	})
```

Comment:
Do you think it makes sense to extend the benchmark tests below to include the selector and expression selector? As this memroy resource client becomes used more widely in our product, it feels like it could be valuable

Reply:
added benchmark tests 3fad32cc61a62e2e02d441c92f16d2e076e967d8
i realized that the existing test wasn't being run (at least not when i ran it locally), probably because `Measure` is deprecated, so i updated to use gmeasure